examples/conversation_mixture_of_agents/conversation_moa.py

```diff
@@ -0,0 +1,157 @@
+        generates a prompt using the current dialogue memory and system
+        prompt, and invokes the language model to produce a response. The
+        response is then formatted and added to the dialogue memory.
+
+        Args:
+            x (`Optional[Union[Msg, Sequence[Msg]]]`, defaults to `None`):
+                The input message(s) to the agent, which also can be omitted if
+                the agent doesn't need any input.
+
+        Returns:
+            `Msg`: The output message generated by the agent.
+        """
+        # record the input if needed
+        if self.memory:
+            self.memory.add(x)
+
+        # # prepare prompt
+        # prompt = self.model.format(
+        #     Msg("system", self.sys_prompt, role="system"),
+        #     self.memory
+        #     and self.memory.get_memory()
+        #     or x,  # type: ignore[arg-type]
+        # )
+        # # call llm and generate response
+        # response = self.model(prompt).text
+
+        # changed the commented model usage above to below:
+        response = self.moa_module(
+            Msg("system", self.sys_prompt, role="system"),
+            self.memory
+            and self.memory.get_memory()
+            or x,  # type: ignore[arg-type]
+        )
+
+        msg = Msg(self.name, response, role="assistant")
+
+        # Print/speak the message in this agent's voice
+        self.speak(msg)
+
+        # Record the message in memory
+        if self.memory:
+            self.memory.add(msg)
+
+        return msg
+
+
+if __name__ == "__main__":
+    # fill you api keys, or host local models using vllm or ollama.
+    model_configs = [
+        {
+            "config_name": "qwen-max",
+            "model_type": "dashscope_chat",
+            "model_name": "qwen-max",
+            "api_key": "{your_api_key}",
+            "generate_args": {
+                "temperature": 0.7,
+            },
+        },
+        {
+            "config_name": "gemini-pro",
+            "model_type": "gemini_chat",
+            "model_name": "gemini-pro",
+            "api_key": "{your_api_key}",
+        },
+        {
+            "config_name": "gpt-4",
+            "model_type": "openai_chat",
+            "model_name": "gpt-4",
+            "api_key": "{your_api_key}",
+            "client_args": {
+                "max_retries": 3,
+            },
+            "generate_args": {
+                "temperature": 0.7,
+            },
+        },
+    ]
+
+    agentscope.init(model_configs=model_configs, project="Mixture of Agents")
+
+    user_agent = UserAgent()
+
+    your_moa_module = MixtureOfAgents(
+        main_model="qwen-max",  # the models you use
+        reference_models=["gpt-4", "qwen-max", "gemini-pro"],
+        show_internal=False,  # set to True to see the internal of MoA modules
+        rounds=1,  # can range from 0 to inf
+    )
+
+    # Init two agents
+    dialog_agent = DialogAgentWithMoA(
+        name="Assistant",
+        sys_prompt="",
```

Comment:
If the agent doesn't need system_prompt, just remove it from the constructor function. Or provide a system_prompt here as `"You're a helpful assistant."`